### PR TITLE
Replace Google Chart with `svg-sparkline.js` for Commit Chart Generation

### PR DESCRIPTION
**Pull request by saadmk11**
closes https://github.com/djangopackages/djangopackages/issues/1128

I was able generate this image URL (DRF Commit Graph) using the script on the PR.

image

> `https://quickchart.io/chart?width=105&height=20&chart=%7B%22type%22%3A%22bar%22%2C%22data%22%3A%7B%22labels%22%3A%5B1%2C1%2C4%2C1%2C4%2C1%2C2%2C2%2C6%2C3%2C0%2C1%2C1%2C1%2C3%2C0%2C1%2C8%2C0%2C1%2C0%2C2%2C0%2C0%2C4%2C2%2C3%2C0%2C0%2C1%2C0%2C0%2C3%2C1%2C0%2C0%2C0%2C4%2C1%2C1%2C5%2C2%2C0%2C1%2C5%2C3%2C4%2C3%2C7%2C10%2C1%2C1%5D%2C%22datasets%22%3A%5B%7B%22data%22%3A%5B1%2C1%2C4%2C1%2C4%2C1%2C2%2C2%2C6%2C3%2C0%2C1%2C1%2C1%2C3%2C0%2C1%2C8%2C0%2C1%2C0%2C2%2C0%2C0%2C4%2C2%2C3%2C0%2C0%2C1%2C0%2C0%2C3%2C1%2C0%2C0%2C0%2C4%2C1%2C1%2C5%2C2%2C0%2C1%2C5%2C3%2C4%2C3%2C7%2C10%2C1%2C1%5D%2C%22backgroundColor%22%3A%22%23666666%22%2C%22borderWidth%22%3A1%7D%5D%7D%2C%22options%22%3A%7B%22legend%22%3A%7B%22display%22%3Afalse%7D%2C%22scales%22%3A%7B%22xAxes%22%3A%5B%7B%22display%22%3Afalse%2C%22gridLines%22%3A%7B%22display%22%3Afalse%7D%7D%5D%2C%22yAxes%22%3A%5B%7B%22display%22%3Afalse%2C%22gridLines%22%3A%7B%22display%22%3Afalse%7D%7D%5D%7D%7D%7D`

**QuickChart Docs:** https://quickchart.io/documentation/
**Community License:** GNU GPL v3 license
**Community Pricing:** $0/month (https://quickchart.io/pricing/)

**Note:** This is a draft experiment, we can choose to use a different service for this as well. (We can always use chartjs directly)

### CODE REVIEW

**Comment by saadmk11** [comment]
I will update the PR if we decide to go with quickchart or we find a better service for this feature. :)

**Comment by jefftriplett** [comment]
In theory, I think it's a good fix. Let me think on the licensing clash, though, and ask around in case I'm thinking that's more of a blocker than it might really be.

**Comment by saadmk11** [comment]
> In theory, I think it's a good fix. Let me think on the licensing clash, though, and ask around in case I'm thinking that's more of a blocker than it might really be.

@jefftriplett I had the same concern. If having the charts as an image is not a requirement, then I think we should give chart.js a try as we are already using chart.js in some parts of the code-base.

I already built an example:

```html
<div style="height:20px; width:105px">
  <canvas class="commitGraph" data-points="1, 7, 5, 15, 4, 4, 11, 11, 2, 12, 9, 4, 10, 1, 3, 11, 9, 9, 3, 2, 8, 13, 2, 4, 0, 6, 14, 5, 12, 8, 13, 12, 2, 3, 6, 9, 6, 13, 3, 9, 8, 10, 4, 4, 2, 3, 9, 11, 5, 3, 13, 13"></canvas>
</div>

<script src="https://cdn.jsdelivr.net/npm/chart.js"></script>

<script>
    function drawCommitGraphs() {
        const elements = document.querySelectorAll(".commitGraph");
        elements.forEach((elm) => {
            const dataPoints = elm.getAttribute("data-points").split(', ');
            drawCommitGraphs(elm, dataPoints);
        });
    }
    function drawCommitGraph(ctx, dataPoints) {
        new Chart(ctx, {
            type: "bar",
            data: {
                labels: dataPoints,
                datasets: [
                    {
                        data: dataPoints,
                        borderWidth: 1,
                        backgroundColor: "#666666",
                    },
                ],
            },
            options: {
                animation: false,
                plugins: {
                    legend: {
                        display: false,
                    },
                    tooltip: {
                        enabled: false,
                    },
                },
                barThickness: 1,
                scales: {
                    x: {
                        display: false,
                    },
                    y: {
                        display: false,
                    },
                },
            },
        });
    }
    drawCommitGraphs()
</script>
```

Let me know your thoughts. :)

**Comment by jefftriplett** [comment]
I'm wondering about the "Rate limit enforced" too from https://quickchart.io/pricing/ 

Do you see any issue with Charts.js on the page that lists ~20 or more?  Example: https://djangopackages.org/categories/apps/

**Comment by jefftriplett** [comment]
I wonder if something like https://github.com/chrisburnell/svg-sparkline would work.

**Comment by ryancheley** [comment]
> I wonder if something like https://github.com/chrisburnell/svg-sparkline would work.

I think this looks like a good alternative. 

The pricing / rate limiting for chart.js may end up biting us if we’re not careful / watching traffic really closely.

**Comment by saadmk11** [comment]
> I wonder if something like https://github.com/chrisburnell/svg-sparkline would work.

Interesting I have never used this before. Will try it out this weekend.

**Comment by saadmk11** [comment]
> I'm wondering about the "Rate limit enforced" too from https://quickchart.io/pricing/
> 
> Do you see any issue with Charts.js on the page that lists ~20 or more? Example: https://djangopackages.org/categories/apps/

Nope, I just created an example but haven't integrated it with the site. I will try it out and compare it with [svg-sparkline](https://github.com/chrisburnell/svg-sparkline) this weekend.

**Comment by saadmk11** [comment]
This is how it looks after the Update:

**Package List:**

Screenshot from 2024-04-17 20-21-40

**Grid:**

Screenshot from 2024-04-17 20-20-57

File: templates/grid/grid_detail.html [hunk 1]
```diff
@@ -291,9 +291,7 @@ <h3>{% trans "Features currently being evaluated" %}</h3>
 {% endblock %}
 
 {% block extra_body %}
-    <script type="text/javascript" src="{{ STATIC_URL }}js/rtd.js"></script>
```
Comment (saadmk11) [inline templates/grid/grid_detail.html:294]: The repo does not have any file called `rtd.js`.

**Comment by jefftriplett** [comment]
@saadmk11 this looks wonderful to me. I will pull it and give it a go now.

**Review by jefftriplett — APPROVED** [review]
✨ great job! This looks good to me.

**Pull request merged**